OpenRA.Game/Network/OrderManager.cs

```diff
@@ -206,6 +221,52 @@ public void Tick()
 			++NetFrameNumber;
 		}
 
+		public void TickPreGame()
+		{
+			SendImmediateOrders();
+
+			ReceiveAllOrdersAndCheckSync();
+
+			Sync.RunUnsynced(Game.Settings.Debug.SyncCheckUnsyncedCode, World, ProcessImmediateOrders);
```

Comment:
`Sync.RunUnsynced` has a large performance overhead (it has to sync the world state twice), so we ideally should have checked that `receivedImmediateOrders` actually had something to avoid doing redundant work when `ProcessImmediateOrders` is a no-op.

Reply:
`RunUnsynced` syncs the world state twice!?

How expensive is this is real terms?